mlflow/store/artifact/azure_blob_artifact_repo.py

```diff
@@ -23,6 +25,9 @@ class AzureBlobArtifactRepository(ArtifactRepository):
     def __init__(self, artifact_uri, client=None):
         super().__init__(artifact_uri)
 
+        _DEFAULT_TIMEOUT = 600  # 10 minutes
```

Comment:
This will be a backwards incompatible change for Azure users, right? Can we require them to set the env var?

Reply:
Based on Azure blob limits, this is quite conservative so I think it's fine